controller/controlplane/controller.go

```diff
@@ -535,82 +365,73 @@ func (r *Reconciler) patchStatus(ctx context.Context, logger logr.Logger, update
 	return ctrl.Result{}, nil
 }
 
-func (r *Reconciler) ensureWebhookResources(
-	ctx context.Context,
-	logger logr.Logger,
-	cp *operatorv1beta1.ControlPlane,
-	enforceConfig bool,
-) (string, op.Result, error) {
-	webhookEnabled := isAdmissionWebhookEnabled(ctx, r.Client, logger, cp)
-	if !webhookEnabled {
-		log.Debug(logger, "admission webhook disabled, ensuring admission webhook resources are not present")
-	} else {
-		log.Debug(logger, "admission webhook enabled, enforcing admission webhook resources")
-	}
-
-	log.Trace(logger, "ensuring admission webhook service")
-	res, admissionWebhookService, err := r.ensureAdmissionWebhookService(ctx, logger, r.Client, cp)
+func (r *Reconciler) generateClientCert(
+	cp types.NamespacedName,
+	caSecret corev1.Secret,
+) ([]byte, []byte, error) {
+	priv, privPem, signatureAlgorithm, err := secrets.CreatePrivateKey(r.ClusterCAKeyConfig)
 	if err != nil {
-		return "", res, fmt.Errorf("failed to ensure admission webhook service: %w", err)
+		return nil, nil, fmt.Errorf("failed to create private key: %w", err)
 	}
-	if res != op.Noop {
-		if !webhookEnabled {
-			log.Debug(logger, "admission webhook service has been removed")
-		} else {
-			log.Debug(logger, "admission webhook service has been created/updated")
-		}
-		return "", res, nil // requeue will be triggered by the creation or update of the owned object
+
+	subject := fmt.Sprintf("%s.%s", cp.Name, cp.Namespace)
+	template := x509.CertificateRequest{
+		Subject: pkix.Name{
+			CommonName:   subject,
+			Organization: []string{"Kong, Inc."},
+			Country:      []string{"US"},
+		},
+		SignatureAlgorithm: signatureAlgorithm,
+		DNSNames:           []string{subject},
 	}
 
-	log.Trace(logger, "ensuring admission webhook certificate")
-	res, admissionWebhookCertificateSecret, err := r.ensureAdmissionWebhookCertificateSecret(ctx, logger, cp, admissionWebhookService)
+	caCertBlock, _ := pem.Decode(caSecret.Data["tls.crt"])
+	if caCertBlock == nil {
+		return nil, nil, errors.New("failed to decode CA certificate")
+	}
+	caCert, err := x509.ParseCertificate(caCertBlock.Bytes)
 	if err != nil {
-		return "", res, err
+		return nil, nil, fmt.Errorf("failed to parse CA certificate: %w", err)
```

Comment:
Let's add some context to this error. I believe mentioning the namespaced name of the CA secret would be very helpful to track down what's causing this error. WDYT?